### PR TITLE
feat: psvExtractor configurable options

### PR DESCRIPTION
**Pull request by carlbrugger**

### CODE REVIEW

File: plugins/psv-extractor/src/psv.extractor.spec.ts [hunk 1]
```diff
@@ -1,56 +1,89 @@
...
+    test('finds the sheet name', () => {
+      const capture = parser.parseBuffer(buffer)
+      const sheetNames = Object.keys(capture)
+      expect(sheetNames).toEqual(['Sheet1'])
+    })
+
+    test('finds the header names', () => {
+      const capture = parser.parseBuffer(buffer)
+      const headers = capture['Sheet1'].headers
+      expect(headers).toHaveLength(4) // Assuming there are 4 headers
```
Comment (hansjhoffman) [inline plugins/psv-extractor/src/psv.extractor.spec.ts:37]: ⛳ This comment doesn't provide much value beyond what the code reads. Maybe we should remove it.

File: plugins/psv-extractor/src/psv.extractor.spec.ts [hunk 2]
```diff
@@ -1,56 +1,89 @@
...
+    })
+
+    test('finds values', () => {
+      const capture = parser.parseBuffer(buffer)
+      const data: Record<string, any> = capture['Sheet1'].data
+      expect(data.length).toBeGreaterThan(0) // Assuming there's at least one row
```
Comment (hansjhoffman) [inline plugins/psv-extractor/src/psv.extractor.spec.ts:43]: ⛳ This comment doesn't provide much value beyond what the code reads. Maybe we should remove it.

File: plugins/psv-extractor/src/psv.extractor.ts [hunk 3]
```diff
@@ -8,45 +8,72 @@ import Papa, { ParseResult } from 'papaparse'
...
+      skipEmptyLines: 'greedy',
+      transform: (value) => value || '',
+      ...options,
+    }
   }
 
-  public parseBuffer(fileContents: string): WorkbookCapture | undefined {
+  public parseBuffer(buffer: Buffer): WorkbookCapture {
     try {
+      const sheetName = 'Sheet1' // Set the sheet name
```
Comment (hansjhoffman) [inline plugins/psv-extractor/src/psv.extractor.ts:41]: ⛳ This comment doesn't provide much value beyond what the code reads. Maybe we should remove it.

File: plugins/psv-extractor/src/psv.extractor.ts [hunk 4]
```diff
@@ -8,45 +8,72 @@ import Papa, { ParseResult } from 'papaparse'
 
 export class PsvExtractor extends AbstractExtractor {
   private readonly _options: {
-    //add if needed
+    delimiter?: ',' | '\t' | '|' | ';'
```
Comment (hansjhoffman) [inline plugins/psv-extractor/src/psv.extractor.ts:11]: ⛳ Personally I think using names here is more readable than symbols. E.g. `delimiter?: "comma" | "tab" | "pipe" | "semicolon"`

File: plugins/psv-extractor/src/psv.extractor.ts [hunk 5]
```diff
@@ -8,45 +8,72 @@ import Papa, { ParseResult } from 'papaparse'
...
+    dynamicTyping?: boolean
+    header?: boolean
```
Comment (hansjhoffman) [inline plugins/psv-extractor/src/psv.extractor.ts:13]: ⛳ I know this is exactly what `papaparse` uses but it's confusing. Meaning "what about the header are you doing?" You don't know unless you pull up the docs. So maybe we could use something like `includeHeader` or anything else you might think that would make this clear to the end user who has no idea what setting this to `true` means to the underlying library.

**Comment by carlbrugger** [comment]
Thanks for the feedback @hansjhoffman! 🌮

**Comment by carlbrugger** [comment]
Close in favor of https://github.com/FlatFilers/flatfile-plugins/pull/141

**Pull request closed without merging**